File: core/src/main/java/com/palomiklo/streampulse/blueprint/IConnectionConfig.java
package com.palomiklo.streampulse.blueprint;

import java.util.function.Supplier;

public interface IConnectionConfig {

    byte getConnectionTimeout();

    byte getInitialDelay();

    byte getPingInterval();

    short getConnectionCleanUpTimeout();

    Supplier<String> getPing();

    Supplier<String> getReconnectEvent();
}

File: core/src/main/kotlin/com/palomiklo/streampulse/blueprint/IConnectionConfig.kt
package com.palomiklo.streampulse.blueprint

import java.util.function.Supplier

interface IConnectionConfig {
    val connectionTimeout: Byte

    val initialDelay: Byte

    val pingInterval: Byte

    val connectionCleanUpTimeout: Short

    val ping: Supplier<String>

    val reconnectEvent: Supplier<String>
}
